### PR TITLE
(656) Check your answers screen

### PR DESCRIPTION
**Pull request by pezholio**
This adds a Check Your Answers Screen to the application and links to it from the task list. This has required a bit of jiggery pokery with the integration tests, so now each integration test Page object has a reference to the TasklistPage Class it's associated with (so, the `CateringPage` has a reference to the `Catering`). This allows us to keep a reference to all the questions that have been answered, as well as the translation, allowing us to assert that the Check Your Answers Page contains references to all the questions that have been answered.

The next jobs are to:

- Ensure that a user can't access CYA until they've answered all the questions
- Add a `Continue` button to the CYA page
- Only allow the form to be submitted when the CYA page has been completed

In the interests of keeping this PR (relatively) lightweight, this will come next

### Screenshots

image

image

### CODE REVIEW

**Comment by pezholio** [comment]
[Check your answers screen](https://trello.com/c/8HjDbWLw/656-check-your-answers-screen)

File: server/controllers/apply/applications/checkYourAnswersController.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import type { Request, Response, RequestHandler, NextFunction } from 'express'
+
+import { ApplicationService } from '../../../services'
+import { sections } from '../../../form-pages/apply'
+
+export default class CheckYourAnswers {
+  constructor(private readonly applicationService: ApplicationService) {}
+
+  show(): RequestHandler {
+    return async (req: Request, res: Response, next: NextFunction) => {
```
Comment (bryangaledxw) [inline server/controllers/apply/applications/checkYourAnswersController.ts:10]: Do we need to take the `next` function here?

File: server/controllers/apply/applications/checkYourAnswersController.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      const application = await this.applicationService.findApplication(req.user.token, req.params.id)
+      const responses = this.applicationService.getResponsesAsSummaryListItems(application)
+
+      res.render(`applications/check-your-answers`, {
```
Comment (bryangaledxw) [inline server/controllers/apply/applications/checkYourAnswersController.ts:14]: Would change to '-quotes

File: cypress_shared/pages/apply/applyPage.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import Page from '../page'
+import TasklistPage from '../../../server/form-pages/tasklistPage'
+import { Application } from '../../../server/@types/shared/models/Application'
+
+import { pages } from '../../../server/form-pages/apply'
+
+export default class ApplyPage extends Page {
+  tasklistPage: TasklistPage
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (bryangaledxw) [inline cypress_shared/pages/apply/applyPage.ts:10]: Is this needed?

File: server/views/applications/check-your-answers.njk [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+              key: {
+                text: "Nationality"
+              },
+              value: {
+                text: application.person.nationality
+              }
+            },
+            {
+              key: {
+                text: "Religion or belief"
```
Comment (bryangaledxw) [inline server/views/applications/check-your-answers.njk:71]: Would capitalise 'B' for consistency with "Gender Identity", "Date of Birth", etc

**Review by bryangaledxw — APPROVED** [review]
Very minor comments, LGTM

File: server/controllers/apply/applications/checkYourAnswersController.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        sections,
+        application,
+        responses,
```
Comment (richpjames) [inline server/controllers/apply/applications/checkYourAnswersController.ts:17]: In most controllers we pass the heading through here, would be good to follow the pattern here

File: server/views/applications/check-your-answers.njk [hunk 6]
```diff
@@ -0,0 +1,134 @@
+{% from "govuk/components/summary-list/macro.njk" import govukSummaryList %}
+{% from "govuk/components/back-link/macro.njk" import govukBackLink %}
+
+{% extends "../partials/layout.njk" %}
+
+{% block pageTitle %}
+ Approved Premises - Check your answers
```
Comment (richpjames) [inline server/views/applications/check-your-answers.njk:7]: We've been using `{% set pageTitle = applicationName + " - " + pageHeading  %}` in the other views
We tend to `{% set mainClasses = "app-container govuk-body" %}` elsewhere but that might not work with this page

**Review by richpjames — APPROVED** [review]
Looks good, I've left a couple comments.

File: server/controllers/apply/applications/checkYourAnswersController.ts (hunk 2 above)
Comment (pezholio) [reply server/controllers/apply/applications/checkYourAnswersController.ts:14]: Weird, I'd have thought Prettier would complain about that!

**Pull request merged**